packages/web/integrations/tfm/router.ts

```diff
@@ -0,0 +1,71 @@
+import { Dec, Int } from "@keplr-wallet/unit";
+import {
+  SplitTokenInQuote,
+  Token,
+  TokenOutGivenInRouter,
+} from "@osmosis-labs/pools";
+
+import { apiClient } from "~/utils/api-client";
+
+import { GetSwapRouteResponse, GetTokensResponse } from "./types";
+
+export class TfmRemoteRouter implements TokenOutGivenInRouter {
+  protected readonly baseUrl: URL;
+
+  constructor(
+    protected readonly osmosisChainId: string,
+    protected readonly tfmBaseUrl: string
+  ) {
+    this.baseUrl = new URL(tfmBaseUrl);
+  }
+
+  async getRoutableCurrencyDenoms(): Promise<string[]> {
+    const params = new URLSearchParams();
+    params.append("isTrading", "True");
+    const queryUrl = new URL(
+      "/api/v1/ibc/chain/osmosis-1/tokens",
+      this.baseUrl.toString()
+    );
```

Comment:
Btw, I believe URL internally has search params available
```suggestion
    const queryUrl = new URL(
      "/api/v1/ibc/chain/osmosis-1/tokens",
      this.baseUrl.toString()
    );
    queryUrl.searchParams.append("isTrading", "True")
```